### PR TITLE
feat: docConfig updated and also added to Target

### PR DESCRIPTION
**Pull request by saadjutt01**
## Issue

[docConfig modifications](https://github.com/sasjs/cli/issues/449)

## Intent

Add docConfig to Target also

## Implementation

New attribute `dataControllerUrl` added to `docConfig`, and added to Target.

## Checks

- [x] Code is formatted correctly (`npm run lint:fix`).
- [x] All unit tests are passing (`npm test`).
- [x] All `sasjs-cli` unit tests are passing (`npm test`).
mocked tests
image

server tests
image

- [x] Reviewer is assigned.

### CODE REVIEW

File: src/types/targetValidators.spec.ts [hunk 1]
```diff
@@ -602,3 +604,25 @@ describe('validateAuthConfig', () => {
     })
   })
 })
+
+describe('validateDocConfig', () => {
+  it('should throw an error when docConfig is null', () => {
+    expect(() =>
+      validateDocConfig((null as unknown) as DocConfig)
+    ).toThrowError('Invalid doc config: JSON cannot be null or undefined.')
+  })
```
Comment (allanbowe) [inline src/types/targetValidators.spec.ts:613]: None of the attributes are mandatory - so am not sure this test is valid?

If the dataControllerUrl is not provided, then we simply shouldn't link to anything (for the tables)

displayMacroCore defaults to true

and outDirectory defaults to sasjsbuild/docs

File: src/types/targetValidators.ts [hunk 2]
```diff
@@ -81,6 +82,14 @@ export const validateAppLoc = (appLoc: string): string => {
   return appLoc
 }
 
+export const validateDocConfig = (docConfig: DocConfig): DocConfig => {
+  if (!docConfig) {
+    throw new Error('Invalid doc config: JSON cannot be null or undefined.')
+  }
```
Comment (allanbowe) [inline src/types/targetValidators.ts:88]: I think it **can** be undefined - it should be possible to generate with a very (very) minimal sasjsconfig.json file.

Check this one out:  https://github.com/sasjs/core/blob/main/sasjs/sasjsconfig.json

(and this one IS used to generate docs, eg the https://core.sasjs.io)

File: src/types/targetValidators.spec.ts (hunk 1 above)
Comment (saadjutt01) [reply src/types/targetValidators.spec.ts:613]: It's checking to have json object if provided.

++ adding default values
```
  if (docConfig.displayMacroCore === undefined) {
    docConfig.displayMacroCore = true
  }

  if (!docConfig.outDirectory) {
    docConfig.outDirectory = ''
  }

  if (!docConfig.dataControllerUrl) {
    docConfig.dataControllerUrl = ''
  }
```

In Cli code, we are using default behavior if "" empty string is present or property is not present

Comment (allanbowe) [reply src/types/targetValidators.spec.ts:613]: aha - ok, great!

**Review by allanbowe — APPROVED** [review]

**Review by allanbowe — APPROVED** [review]

**Pull request merged**